src/neo-vm/ExecutionEngine.cs

```diff
@@ -1227,52 +1227,37 @@ private bool ExecuteEndTry(int endOffset)
 
         private bool HandleException()
         {
-            foreach (var executionContext in InvocationStack)
+            while (InvocationStack.TryPeek(out var executionContext))
             {
-                if (executionContext.TryStack is null) continue;
-
-                while (executionContext.TryStack.TryPeek(out ExceptionHandingContext tryContext))
+                if (executionContext.TryStack != null)
                 {
-                    if (tryContext.State == ExceptionHandingState.Finally || (tryContext.State == ExceptionHandingState.Catch && !tryContext.HasFinally))
-                    {
-                        executionContext.TryStack.Pop();
-                        continue;
-                    }
-
-                    ResumeContext(tryContext);
-                    if (tryContext.State == ExceptionHandingState.Try && tryContext.HasCatch)
-                    {
-                        tryContext.State = ExceptionHandingState.Catch;
-                        Push(UncaughtException);
-                        CurrentContext.InstructionPointer = tryContext.CatchPointer;
-                        UncaughtException = null;
-                    }
-                    else
+                    while (executionContext.TryStack.TryPeek(out var tryContext))
                     {
-                        tryContext.State = ExceptionHandingState.Finally;
-                        CurrentContext.InstructionPointer = tryContext.FinallyPointer;
+                        if (tryContext.State == ExceptionHandingState.Finally || (tryContext.State == ExceptionHandingState.Catch && !tryContext.HasFinally))
+                        {
+                            executionContext.TryStack.Pop();
+                            continue;
+                        }
+                        if (tryContext.State == ExceptionHandingState.Try && tryContext.HasCatch)
+                        {
+                            tryContext.State = ExceptionHandingState.Catch;
+                            Push(UncaughtException);
+                            CurrentContext.InstructionPointer = tryContext.CatchPointer;
+                            UncaughtException = null;
+                        }
+                        else
+                        {
+                            tryContext.State = ExceptionHandingState.Finally;
+                            CurrentContext.InstructionPointer = tryContext.FinallyPointer;
+                        }
+                        return true;
                     }
-
-                    return true;
                 }
+                ContextUnloaded(InvocationStack.Pop());
             }
             return false;
         }
 
-        private void ResumeContext(ExceptionHandingContext context)
-        {
```

Comment:
I think we need `ResumeContext`, as when some exception happen without try-catch, we should reserve `InvocationStack`.

Reply:
Fixed.